### PR TITLE
Add way to set release, dist, environment, serverName, tags, and extr…

### PR DESCRIPTION
**Pull request by bretthoerner**
…aTags via the DSN.

### CODE REVIEW

**Comment by bretthoerner** [comment]
Need to add docs but this is ready for review otherwise.

**Comment by bretthoerner** [comment]
Oh, the plan after this is (in theory) to remove all of these fields from the appenders, and instead have appender configuration setup the client fields. Making a different branch/PR so I don't blow this up into a huge one like the others.

**Comment by bretthoerner** [comment]
TODO: Better document tags vs extraTags

File: sentry/src/main/java/io/sentry/SentryClient.java [hunk 1]
```diff
@@ -147,6 +151,10 @@ public void sendEvent(EventBuilder eventBuilder) {
             eventBuilder.withTag(tagEntry.getKey(), tagEntry.getValue());
         }
 
+        // Note that extraTags are missing here because they only exist to be used by
+        // other systems (before this method is even called). For example: the logging
+        // integrations check the SLF4J MDC system for extras.
```
Comment (tkaemming) [inline sentry/src/main/java/io/sentry/SentryClient.java]: I don't understand what this comment is trying to say.

Comment (bretthoerner) [inline sentry/src/main/java/io/sentry/SentryClient.java]: Eh ya, it started as a note to self about why all the other fields were touched here and I think it turned out far more confusing than helpful.

File: docs/config.rst [hunk 2]
```diff
@@ -84,7 +84,7 @@ useful inside shared application containers),
 
 ::
 
-    http://public:private@host:port/1?http.proxy.host=proxy.example.com&sentry.http.proxy.port=8080
+    http://public:private@host:port/1?http.proxy.host=proxy.example.com&http.proxy.port=8080
```
Comment (bretthoerner) [inline docs/config.rst]: I missed some `sentry.` previously, cleaned those up here.

**Comment by bretthoerner** [comment]
Yeah it's weird because the MDC thing is the only usage now, otherwise it's ignored, but future (or custom) code may do something different with it.

**Pull request merged**